draft-kwbdgrr-tsvwg-net-collab-rqmts.md

```diff
@@ -396,6 +396,52 @@ For the requirements that follow, the assumption is that the client agrees to th
 REQ-PACKET-SELF: Packet importance is indicated by the packet itself, which may need to be decrypted or de-obfuscated.
 
 
+## Host-Network Metadata {#host-network}
+
+### Priority between Flows (Inter-flow) {#interflow-priority}
+
+Certain flows being received by a host (or by an application on a host) are less or more important than other flows of *the same host*.
+For example, a host downloading a software update is generally considered less important than another host doing interactive audio/video or gaming.
+By signaling the relative importance of flows to a network element, the network element can (de-)prioritize those flows to best accommodate the needs of the various applications on a same host.
+
+Without a signaling in place between a receiving host and its network, remote peers are able to mark packets that interfere with the desires of the receiving host -- making their flows more important than what the receiving host considers more important.
+This eventually causes all flows to be marked as important, or -- more likely -- such priority markings to be ignored.
+
+However, prioritizing between flows presents challenges because the host can have both malicious and legitimate applications, and the remote peers can also be malicious and benign.
```

Comment:
Should we mention that this is out of the scope of the document, and we only discuss intra-flow? If we are not focusing on inter-flow, do we need a separate section for it? I was thinking of mentioning what inter-flow is and that is out of the scope in a single in as part of intra-flow and remove the whole section here so that it avoids any confusion at all (c.f https://github.com/SpencerDawkins/net-collab-rqmts/pull/27/files).

Reply:
I do see a value in keeping the text in the first version of the draft but with reasonable caveats such as be restricted to flows bound to the same user/host.

We can limit the scope to cases where an app can lower its preferences and the like.

Reply:
same user/host would be outside of the same 5-tuple right? If that's the case, there are chances of application running in user mode to be hooked into and priority can be lowered by a malware resulting in a DoS. I fear there might be some complications there, at least in terms of implementation, if we consider outside of the 5-tuple.

Reply:
There are complications. But there are ways to associate the 5-tuples with each other by demonstrating knowledge of something that is in the clear, like the port number, or sending a STUN packet (on that same 5-tuple), or in the flow's initial packet (e.g., QUIC long header in the initial QUIC exchange), or using the same TLS Grease value for all of the associated flows...  Lots of methods can be considered.

Reply:
Fully agree there are complications and there might be ways to soften some of them. We don't need to dig into those now, but leave that as a discussion item to have in the WG. Thanks